test/Microsoft.Azure.WebJobs.Host.EndToEndTests/Scale/ScaleHostEndToEndTests.cs

```diff
@@ -0,0 +1,379 @@
+    {
+        public static IHostBuilder ConfigureTestExtensionBScale(this IHostBuilder builder, TriggerMetadata triggerMetadata)
+        {
+            builder.ConfigureServices((context, services) =>
+            {
+                services.AddSingleton<IScaleMonitorProvider>(serviceProvider =>
+                {
+                    IConfiguration config = serviceProvider.GetService<IConfiguration>();
+                    IOptions<ScaleOptions> scaleOptions = serviceProvider.GetService<IOptions<ScaleOptions>>();
+                    return new TestExtensionBScalerProvider(config, scaleOptions, triggerMetadata);
+                });
+                services.AddSingleton<ITargetScalerProvider>(serviceProvider =>
+                {
+                    IConfiguration config = serviceProvider.GetService<IConfiguration>();
+                    IOptions<ScaleOptions> scaleOptions = serviceProvider.GetService<IOptions<ScaleOptions>>();
+                    return new TestExtensionBScalerProvider(config, scaleOptions, triggerMetadata);
+                });
+            });
+
+            return builder;
+        }
+
+        private class TestExtensionBScalerProvider : TestExtensionScalerProvider
+        {
+            public TestExtensionBScalerProvider(IConfiguration config, IOptions<ScaleOptions> scaleOptions, TriggerMetadata triggerMetadata)
+                : base(config, scaleOptions, triggerMetadata)
+            {
+            }
+        }
+    }
+
+    internal class TestScaleMonitor : IScaleMonitor
+    {
+        public ScaleMonitorDescriptor Descriptor { get; set; }
+
+        public TestScaleMonitor(string id, string functionId)
+        {
+            Descriptor = new ScaleMonitorDescriptor(id, functionId);
+        }
+
+        public Task<ScaleMetrics> GetMetricsAsync()
+        {
+            return Task.FromResult(new ScaleMetrics());
+        }
+
+        public ScaleStatus GetScaleStatus(ScaleStatusContext context)
+        {
+            int targetWorkerCount = (int)Char.GetNumericValue(Descriptor.FunctionId[Descriptor.FunctionId.Length - 1]);
+
+            return new ScaleStatus()
+            {
+                Vote = ScaleVote.ScaleOut
+            };
+        }
+    }
+
+    internal class TestTargetScaler : ITargetScaler
+    {
+        public TargetScalerDescriptor TargetScalerDescriptor { get; set; }
+
+        public TestTargetScaler(string functionId)
+        {
+            TargetScalerDescriptor = new TargetScalerDescriptor(functionId);
+        }
+
+        public Task<TargetScalerResult> GetScaleResultAsync(TargetScalerContext context)
+        {
+            int targetWorkerCount = (int)Char.GetNumericValue(TargetScalerDescriptor.FunctionId[TargetScalerDescriptor.FunctionId.Length - 1]) + 1;
+
+            return Task.FromResult(new TargetScalerResult()
+            {
+                TargetWorkerCount = targetWorkerCount
+            });
+        }
+    }
+
+    internal class FakeAzureComponentFactory
```

Comment:
Remove if not needed

Reply:
on a related note, how is Managed Identity supported in this new model? I remember in the old version of the API the AzureComponentFactory was injected

Reply:
we are using `TriggerData.Properties` to pass `AzureComponentFactory`:
https://github.com/Azure/azure-sdk-for-net/pull/34557

Reply:
we are using `TriggerData.Properties` to pass `AzureComponentFactory`:
https://github.com/Azure/azure-sdk-for-net/pull/34557